ecommerce/frontend/src/components/ChooseListModal.js

```diff
@@ -0,0 +1,54 @@
+import './ChooseListModal.less'
+
+import React, { useEffect, useState } from 'react'
+import { List } from 'antd'
+import { sleep } from '../utils'
+
+export const ChooseListModal = ({ onChooseList }) => {
+    const [lists, setLists] = useState([])
+    const [loading, setLoading] = useState(true)
+
+    useEffect(() => {
+        async function fetch() {
+            try {
+                setLoading(true)
+                await sleep(1000)
+
+                setLists([
+                    {
+                        id: 1,
+                        name: 'Electronics',
+                    },
+
+                    {
+                        id: 2,
+                        name: 'Thanksgiving',
+                    },
+                    {
+                        id: 3,
+                        name: 'Blackfriday',
+                    },
+                ])
+            } catch (error) {
+            } finally {
+                setLoading(false)
+            }
+        }
+
+        fetch()
+    }, [])
+
+    const onListClick = list => () => onChooseList(list)
+
+    return (
+        <List loading={loading} size="large">
+            {lists.map((list, ix) => {
+                return (
+                    <List.Item className={'choose-list-modal-list'} onClick={onListClick(list)} key={list.id ?? ix}>
+                        {list.name ?? `Example list name ${ix}`}
```

Comment:
Maybe instead of selecting a single list, we can have a checkbox display so that user can add the product into multiple lists at the same request, not a must but it's a design choice.

Reply:
We should discuss this, but usual use case is to add to single list, like in Facebook for example for saved links/videos. Also people usually have a list for an event such as "Birthday" or "Black friday" or "Christmas" things like that.